loki/transformations/loop_blocking.py

```diff
@@ -0,0 +1,310 @@
+                                                 name=loop.variable.name + "_loop_block_end")))
+
+    routine.variables += splitting_vars.splitting_vars
+
+    # block index calculations
+    blocking_body = (
+        Assignment(splitting_vars.block_start,
+                   parse_expr(
+                       f"({splitting_vars.block_idx} - 1) * {splitting_vars.block_size} + 1")
+                   ),
+        Assignment(splitting_vars.block_end,
+                   InlineCall(DeferredTypeSymbol('MIN', scope=routine),
+                              parameters=(Product(children=(
+                                  splitting_vars.block_idx, splitting_vars.block_size)),
+                                          loop.bounds.upper))
+                   ))
+
+    # Outer loop blocking variable assignments
+    loop_range = loop.bounds
+    block_loop_inits = (
+                        ir.Assignment(splitting_vars.num_blocks,
+                                      ceil_division(num_iterations(loop_range),
+                                                    splitting_vars.block_size)),
+    )
+
+    # Inner loop
+    iteration_nums = (
+        Assignment(splitting_vars.iter_num,
+                   parse_expr(f"{splitting_vars.block_start}+{splitting_vars.inner_loop_var}-1")),
+        Assignment(loop.variable,
+                   iteration_index(splitting_vars.iter_num, loop_range))
+    )
+    inner_loop = Loop(variable=splitting_vars.inner_loop_var, body=iteration_nums + loop.body,
+                      bounds=LoopRange(
+                          (IntLiteral(1), parse_expr(
+                              f"{splitting_vars.block_end} - {splitting_vars.block_start} + 1"))))
+
+    #  Outer loop bounds + body
+    outer_loop = Loop(variable=splitting_vars.block_idx, body=blocking_body + (inner_loop,),
+                      bounds=LoopRange((IntLiteral(1), splitting_vars.num_blocks)))
+    change_map = {loop: block_loop_inits + (outer_loop,)}
+    Transformer(change_map, inplace=True).visit(routine.ir)
+    return splitting_vars, inner_loop, outer_loop
+
+
+def blocked_shape(a: Array, blocking_indices, block_size):
+    """
+    calculates the dimensions for a blocked version of the array.
+    """
+    shape=tuple(
+        IntLiteral(block_size) if isinstance(dim, Scalar) and any(
+            bidx in dim for bidx in blocking_indices) else dim for dim
+        in a.shape)
+    return shape
+
+
+def blocked_type(a: Array):
+    return a.type.clone(intent=None)
+
+
+def replace_indices(dimensions, indices: list, replacement_index):
+    """
+    Returns a new dimension object with all occurences of indices changed to replacement_index.
+
+    Parameters
+    ----------
+    dimensions: Symbolic representation of dimensions or indices.
+    indices: list of `Variable`s that will be replaced in the new :any:`Dimension` object.
+    replacement_index: :any:`Expression` replacement for the indices changed.
```

Comment:
Unfortunately, Sphinx expects the Parameters description in the following format:
```
<arg_name> : <arg type>
    <description>
```

Otherwise, the formatting goes wrong, see here: https://sites.ecmwf.int/docs/loki/362/loki.transformations.loop_blocking.html#loki.transformations.loop_blocking.replace_indices